eZ/Publish/Core/Persistence/Legacy/User/Handler.php

```diff
@@ -166,35 +167,80 @@ public function delete($userId)
     }
 
     /**
-     * Create new role.
+     * Create new role draft.
      *
-     * @param \eZ\Publish\SPI\Persistence\User\Role $role
+     * Sets status to Role::STATUS_DRAFT on the new returned draft.
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\RoleCreateStruct $createStruct
      *
      * @return \eZ\Publish\SPI\Persistence\User\Role
      */
-    public function createRole(Role $role)
+    public function createRole(RoleCreateStruct $createStruct)
     {
+        return $this->internalCreateRole($createStruct);
+    }
+
+    /**
+     * Creates a draft of existing defined role.
+     *
+     * Sets status to Role::STATUS_DRAFT on the new returned draft.
+     *
+     * @param mixed $roleId
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If role with defined status is not found
+     *
+     * @return \eZ\Publish\SPI\Persistence\User\Role
+     */
+    public function createRoleDraft($roleId)
+    {
+        $createStruct = $this->mapper->createCreateStructFromRole(
+            $this->loadRole($roleId)
+        );
+        $createStruct->status = Role::STATUS_DRAFT;
+
+        return $this->internalCreateRole($createStruct, $roleId);
+    }
+
+    /**
+     * Internal method for creating Role.
+     *
+     * Used by self::createRole() and self::createRoleDraft()
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\RoleCreateStruct $createStruct
+     * @param mixed|null $roleId Used by self::createRoleDraft() to retain Role id in the draft
+     *
+     * @return \eZ\Publish\SPI\Persistence\User\Role
+     */
+    protected function internalCreateRole(RoleCreateStruct $createStruct, $roleId = null)
+    {
+        $createStruct = clone $createStruct;
+        $role = $this->mapper->createRoleFromCreateStruct(
+            $createStruct
+        );
+        $role->id = $roleId;
+
         $this->roleGateway->createRole($role);
 
         foreach ($role->policies as $policy) {
-            $this->addPolicy($role->id, $policy);
+            $this->addPolicyByRoleDraft($role->id, $policy);
         }
 
         return $role;
     }
 
     /**
-     * Loads a specified role by $roleId.
+     * Loads a specified role (draft) by $roleId.
```

Comment:
`Loads a role by $roleId and $status.`

Below:
`@throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If role with given status does not exist`

You should also return that information in the exception, see similar: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Persistence/Legacy/Exception/TypeNotFound.php